src/shared_realm.cpp

```diff
@@ -451,6 +451,14 @@ void Realm::update_schema(Schema schema, uint64_t version, MigrationFunction mig
     }
 
     if (initialization_function && old_schema_version == ObjectStore::NotVersioned) {
+        // Initialization function needs to see the latest schema
+        uint64_t temp_version = ObjectStore::get_schema_version(read_group());
```

Comment:
it is not always `NotVersioned`, it will be set to the version when the schema & version has been initialized.